pkg/controller/dual-pods/server-requesting.go

```diff
@@ -133,6 +124,13 @@ func (ctl *controller) processServerRequestingPod(ctx context.Context, requestin
 		return nil, false
 	}
 
+	if node.Spec.Unschedulable {
+		// Reflect the inability to serve back to the client/user
+		logger.V(2).Info("Deleting server-requesting Pod because it is bound to an unschedulable Node and has no server-running Pod", "pod", requestingPod.Name, "node", requestingPod.Spec.NodeName)
+		err := ctl.coreclient.Pods(requestingPod.Namespace).Delete(ctx, requestingPod.Name, metav1.DeleteOptions{PropagationPolicy: ptr.To(metav1.DeletePropagationBackground)})
```

Comment:
Requesting pods are not owned by the dual-pods controller and shouldn't be deleted.